firmware/src/rdm_util.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * rdm_util.c
+ * Copyright (C) 2015 Simon Newton
+ */
+#include "rdm_util.h"
+
+#ifdef HAVE_ARPA_INET_H
+#include <arpa/inet.h>
+#endif
+
+#include "system_pipeline.h"
+#include "constants.h"
+#include "utils.h"
+
+bool RDMUtil_RequiresAction(const RDMResponder *responder,
+                            const uint8_t uid[UID_LENGTH]) {
+  if (RDMUtil_UIDCompare(responder->uid, uid) == 0) {
+    return true;
+  }
+
+  if (uid[2] != 0xff || uid[3] != 0xff || uid[4] != 0xff || uid[5] != 0xff) {
```

Comment:
It's been refactored. It still exists in the old location because I'm sending these in small batches.